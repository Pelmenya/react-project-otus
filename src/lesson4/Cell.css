.cell {
  width: 15px;
  height: 15px;
  border: 1px solid #000;
  display: inline-flex;
  margin: 1px;
  text-align: center;
  box-sizing: border-box;
}

.cell_dead {
  background-color: #fff;
}

.cell_live {
  background-color: #00ff00;
}

