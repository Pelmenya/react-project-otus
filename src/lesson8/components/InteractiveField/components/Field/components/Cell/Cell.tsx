import React, { FC } from "react";
import styled from "@emotion/styled";
import { css } from "@emotion/core";

import type { CellProps } from "types/field";

const CellClass = css`
  width: 15px;
  height: 15px;
  border: 1px solid #000;
  display: inline-flex;
  margin: 1px;
  text-align: center;
  box-sizing: border-box;
`;

const CellDeadClass = css`
  background-color: #fff;
`;

const CellLiveClass = css`
  background-color: #00ff00;
`;

interface Props {
  isFilled: boolean;
}

export const CellItem = styled.button`
  ${CellClass};
  ${({ isFilled }: Props) => (isFilled ? CellLiveClass : CellDeadClass)};
`;

export const Cell: FC<CellProps> = ({ filled, x, y, onClick }) => {
  const isFilled = Boolean(filled);
  return (
    <CellItem isFilled={isFilled} onClick={() => onClick(x || 0, y || 0)}>
      {filled}
    </CellItem>
  );
};
