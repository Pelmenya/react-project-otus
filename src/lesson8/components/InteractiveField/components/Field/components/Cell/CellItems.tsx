import styled from "@emotion/styled";
import { css } from "@emotion/core";

const Cell = css`
  width: 15px;
  height: 15px;
  border: 1px solid #000;
  display: inline-flex;
  margin: 1px;
  text-align: center;
  box-sizing: border-box;
`;

const CellDead = css`
  background-color: #fff;
`;

const CellLive = css`
  background-color: #00ff00;
`;

interface Props {
  isFilled: boolean;
}

export const CellItem = styled.button`
  ${Cell};
  ${({ isFilled }: Props) => (isFilled ? CellLive : CellDead)};
`;
