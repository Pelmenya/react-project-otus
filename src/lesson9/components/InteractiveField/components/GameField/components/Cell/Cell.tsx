import React, { FC } from "react";
import styled from "@emotion/styled";
import { css } from "@emotion/core";

import type { CellProps } from "types/field";

const CellClass = css`
  width: 15px;
  height: 15px;
  border: 1px solid #000;
  display: inline-flex;
  text-align: center;
  box-sizing: border-box;
  cursor: pointer;
  text-align: center;
  vertical-align: top;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 2px;
  padding-bottom: 0px;
  font-size: 10px;
  line-height: 1;
  transition: transform 1s, background-color 1s;
`;

const CellDeadClass = css`
  background-color: #fff;
`;

const CellLiveClass = css`
  background-color: #00ff00;
  transform: rotate(360deg);
`;

interface Props {
  isFilled: boolean;
}

const CellItem = styled.button`
  ${CellClass};
  ${({ isFilled }: Props) => (isFilled ? CellLiveClass : CellDeadClass)};
`;

export const Cell: FC<CellProps> = ({ filled, x, y, onClick }) => {
  const isFilled = Boolean(filled);
  return (
    <CellItem isFilled={isFilled} onClick={() => onClick(x || 0, y || 0)}>
      {filled}
    </CellItem>
  );
};
