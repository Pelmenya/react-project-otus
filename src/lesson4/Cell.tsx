import React, { FC } from "react";
import "./Cell.css";
import type { CellProps } from "./interfaces";

/**
 * Some custom description for Cell component
 */
const Cell: FC<CellProps> = ({ filled, x, y, onClick }) => {
  if (filled) {
    return <button className="cell cell_live">{filled}</button>;
  }
  return (
    <button className="cell cell_dead" onClick={() => onClick(x || 0, y || 0)}>
      {filled}
    </button>
  );
};

export default Cell;

export function getCell(props: CellProps) {
  // На самом деле это означает тоже самое что и
  // return <Cell x={props.x} y={props.y} onClick={props.onClick} />;
  // Все параметры передаются в компонент
  return <Cell {...props} />;
}
