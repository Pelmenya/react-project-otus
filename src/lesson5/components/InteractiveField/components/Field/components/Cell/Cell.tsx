import React, { FC } from "react";
import type { CellProps } from "src/lesson8/components/InteractiveField/components/Field/node_modules/types/field";

import { CellItem } from "./CellItems";

export const Cell: FC<CellProps> = ({ filled, x, y, onClick }) => {
  const isFilled = Boolean(filled);
  return (
    <CellItem isFilled={isFilled} onClick={() => onClick(x || 0, y || 0)}>
      {filled}
    </CellItem>
  );
};
