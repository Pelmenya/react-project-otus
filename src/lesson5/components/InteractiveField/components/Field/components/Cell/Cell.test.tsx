import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";

import { Cell } from "./Cell";

describe("Cell", () => {
  it("renders button for dead cell", () => {
    expect(
      renderer.create(<Cell onClick={jest.fn()} />).toJSON()
    ).toMatchSnapshot();
  });

  it("renders button for live cell", () => {
    const isFilled = " ";
    expect(
      renderer.create(<Cell filled={isFilled} onClick={jest.fn()} />).toJSON()
    ).toMatchSnapshot();
  });

  it("calls onClick callback on click by dead cell", () => {
    const onClick = jest.fn();
    const wrapper = mount(<Cell onClick={onClick} />);
    wrapper.simulate("click");
    expect(onClick).toHaveBeenCalled();
  });

  it("calls onClick callback on click by live cell", () => {
    const onClick = jest.fn();
    const isFilled = " ";
    const wrapper = mount(<Cell filled={isFilled} onClick={onClick} />);
    wrapper.simulate("click");
    expect(onClick).toHaveBeenCalled();
  });

  it("calls onClick callback with passed x, y params", () => {
    const onClick = jest.fn();
    const x = 12;
    const y = 14;
    const wrapper = mount(<Cell onClick={onClick} x={x} y={y} />);
    wrapper.simulate("click");
    expect(onClick).toHaveBeenCalledWith(x, y);
  });
});
