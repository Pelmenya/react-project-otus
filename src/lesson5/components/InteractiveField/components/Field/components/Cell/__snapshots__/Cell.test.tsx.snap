// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell renders button for dead cell 1`] = `
.emotion-0 {
  width: 15px;
  height: 15px;
  border: 1px solid #000;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: 1px;
  text-align: center;
  box-sizing: border-box;
  background-color: #fff;
}

<button
  className="emotion-0"
  onClick={[Function]}
/>
`;

exports[`Cell renders button for live cell 1`] = `
.emotion-0 {
  width: 15px;
  height: 15px;
  border: 1px solid #000;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: 1px;
  text-align: center;
  box-sizing: border-box;
  background-color: #00ff00;
}

<button
  className="emotion-0"
  onClick={[Function]}
>
   
</button>
`;
